test/e2e/shoot/create_update_delete.go

```diff
@@ -62,6 +65,14 @@ var _ = Describe("Shoot Tests", Label("Shoot", "default"), func() {
 			g.Expect(shootClient.Client().List(ctx, &corev1.NamespaceList{})).To(Succeed())
 		}).Should(Succeed())
 
+		By("Verify the shoot event-logging")
+		Eventually(func(g Gomega) {
+			eventLoggerVerifier.Before(ctx)
+			eventLoggerVerifier.Prepare(ctx)
+			eventLoggerVerifier.Expect(ctx)
+			eventLoggerVerifier.After(ctx)
```

Comment:
I don't think it makes sense to artificially press this verify into this `Before`, `Prepare`, ... structure. We have used it for the credentials rotation because it makes sense there.
But you can simplify this here and put everything into a single `Verify` method?

Reply:
As you wish. I just followed your advice to make a vast interface as the in the example which you gave me.